core/controller/src/main/scala/whisk/core/controller/Triggers.scala

```diff
@@ -129,7 +129,7 @@ trait WhiskTriggersApi extends WhiskCollectionAPI {
 
           if (activeRules.nonEmpty) {
             val args: JsObject = trigger.parameters.merge(payload).getOrElse(JsObject())
-            val actionLogList: Iterable[Future[JsObject]] = activateRules(user, args, activeRules)
+            val actionLogList: Iterable[Future[JsObject]] = activateRules(user, args, trigger.rules.get)
```

Comment:
I can't be absolutely certain here that trigger.rules.get is safe.
@markusthoemmes in your refactoring, I think this becomes avoidable. Should I merge that first?

Reply:
Just go with a `getOrElse(Map.empty)`. `activateRules` just won